### PR TITLE
certmgr: Add patch for optional trust of self-signed certificates at remote cfssl apiserver

### PR DESCRIPTION
**Pull request by johanot**
###### Motivation for this change

Without this patch, it is not possible to use a self-signed certificate for a remote cfssl server without certmgr rejecting the cert as untrusted. 

This patch allows for the user to configure (optionally) a trusted CA-cert as part of any certmgr cert-spec.

See also: https://github.com/cloudflare/certmgr/pull/51

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by srhb** [comment]
While I realize that it's an unfortunate situation that this isn't merged yet, since the revamped k8s module depends on it, I think we should have at least *some* indication that upstream is going to actually be merging this, and not just a fetchpatch from an open PR, before we include it under the "certmgr" name and not a clearly forked version.

**Comment by vielmetti** [comment]
I'll also drop in a reference to Jetstack's cert manager which would appear to support the self-signed certs, based on https://github.com/jetstack/cert-manager/issues/84 and https://github.com/jetstack/cert-manager/pull/637 -- assuming that the broader requirements could potentially be met by that effort as well.

**Comment by fpletz** [comment]
To get the kubernetes refactor merged for 19.03, I've added a separate attribute for the certmgr package with the patch and added a package option to the certmgr service. Also I've rebased on top of current master.

File: pkgs/tools/security/certmgr/default.nix [hunk 1]
```diff
@@ -1,23 +1,43 @@
...
-    license = licenses.bsd2;
-    maintainers = with maintainers; [ johanot srhb ];
+  certmgr-selfsigned = generic {
+    # The following patch makes it possible to use a self-signed x509 cert
+    # for the cfssl apiserver.
+    # TODO: remove patch when PR is merged.
+    patches = [
+      (fetchpatch {
+        url    = "https://github.com/cloudflare/certmgr/pull/51.patch";
+        sha256 = "0jhsw159d2mgybvbbn6pmvj4yqr5cwcal5fjwkcn9m4f4zlb6qrs";
```
Comment (volth) [inline pkgs/tools/security/certmgr/default.nix:39]: `sha256` has changed.
Patch anyway merged upstream, so it can be removed

**Pull request merged**